roles/dhcp-server/templates/pxe.conf.j2

```diff
@@ -0,0 +1,24 @@
+# /etc/dnsmasq.d/pxe.conf
+
+# https://wiki.fogproject.org/wiki/index.php?title=BIOS_and_UEFI_Co-Existence#Using_ProxyDHCP_.28dnsmasq.29
+# http://forum.ipxe.org/showthread.php?tid=6077
+
+# Tag requests from ipxe client
+dhcp-match=set:ipxe,175 # iPXE sends a 175 option.
+
+# tag/respond to legacy bios
+dhcp-match=set:bios,60,PXEClient:Arch:00000
+dhcp-boot=tag:bios,ipxe/undionly.kpxe #,dc10b,192.168.1.4
+dhcp-boot=tag:bios,tag:ipxe,boot/grub/i386-pc/core.0 #,dc10b,192.168.1.4
+
+# tag to efi
+dhcp-match=set:efibc,60,PXEClient:Arch:00007
+
+# Chain load ipxe, then grubnetx64.efi
+# (doesn't work. grub doesn't request any files from tftp, like boot/grub.cfg so #out )
+# dhcp-boot=tag:efibc,ipxe.efi.signed,dc10b,192.168.1.4
+# dhcp-boot=tag:efibc,tag:ipxe,grubnetx64.efi.signed,dc10b,192.168.1.4
```

Comment:
I'd rather see something that doesn't work yet not included than commented out.

Reply:
I'd like to leave this in. "this" works, but current version of grub does not.
yeah, cutting words on what "doesnt work" means.

If someone (like future me) wants to implement this, they probably want to know that this is it, but it doesn't work.  
someone in #grub said it should work, we suspect there is a bug but I didn't feel like perusing it.
so maybe grub will get magically fixed and this will be in place.

Reply:
let's uncomment it then?

Reply:
uncommenting will break a working eufi install.  

the comments are a workaround - likely a grub bug.  

let me splain:
legacy pxe rom -> ipxe -> grub-net (menu) -> installer 
eufi pxe rom -> grub-net (menu) -> installer 
no eufi ipxe, but works.

ipxe is very hand for debugging.  as it is booting it promps "^b for command line" where you can do dhcp requests and inspect and modify all the things.   server IP, next-server IP, file name, tftp gets.
For production it needlessly adds 5 seconds to the process.

If we A) do need uefi and B) have problems where C) someone wants to add uefi ipxe to debug, step 1 will be uncomment and see if it works.  step 2 figure out why it doesn't.    Shave the Yac.

Reply:
Have a chainload_ipxe flag?

Reply:
two problems:
1. ipxe works for legacy bios. which is currently everything I have come across.  I don't want that turned off.

2. To me having a formal option like that implies both options work.

I  wonder if you understand the state thins are now.  which includes my lack of understanding what the problem is and my not caring because I currently don't need it.

the flag would be 
enable_bugs=False ;)

or maybe 
workaround_mystery_problem = True

but once the problem is solved, the option is no longer needed and it becomes cruft that should be removed: both the default option and the {% if option.. code in the template.  

it is a feature someone (me) may want, but it needs work.  
As is there is a little bit of knowledge stored there.  Which is what comments are for, right? 
it is more than just #commented out code, it is code that I feel comfortable saying should work, but for some reason it doesn't.   If all that gets removed, the next person that goes down that path will have to re-invent it, and then likely burn up time trying to figure out why what should work doesn't work.